### ISSUE TITLE
Audit required for insecure usage of OSCrypt

### ISSUE THREAD

**Issue by orspetol** [brave/brave-browser#15127]
## Description 
`OSCrypt::EncryptString()` reuses the [same IV in CBC mode](https://chromium.googlesource.com/chromium/src/+/lkgr/components/os_crypt/os_crypt_linux.cc#178) for the cipher. This is a chromium concern (and not necessarily a bug); however, there may be use cases in Brave where this can create a privacy violation for our users (as well as other times where this is perfectly okay). 

What does this mean? If we reuse the value (plaintext) for any reason in the `encrypstring` method, a malicious party will be able to distinguish if it has been reused under the same key. 

The use of `OSCrypt::EncryptString()` should be carefully reviewed to ensure out user's data is not experiencing a privacy violation.

**Comment by bsclifton** [comment]
cc: @diracdeltas - I don't know if this was a former employee who filed? I don't understand enough about this to take any action. Can you help triage?

**Comment by diracdeltas** [comment]
yes, it was by a former employee and i think we should leave it open as p4.
